BQ25895M.device.nut

```diff
@@ -0,0 +1,302 @@
+//Registers Addresses
+const BQ25895M_REG00 = 0x00;
+const BQ25895M_REG01 = 0x01;
+const BQ25895M_REG02 = 0x02;
+const BQ25895M_REG03 = 0x03; 
+const BQ25895M_REG04 = 0x04;
+const BQ25895M_REG05 = 0x05;
+const BQ25895M_REG06 = 0x06;
+const BQ25895M_REG07 = 0x07;
+const BQ25895M_REG08 = 0x08;
+const BQ25895M_REG09 = 0x09;
+const BQ25895M_REG0A = 0x0A;
+const BQ25895M_REG0B = 0x0B;
+const BQ25895M_REG0C = 0x0C;
+const BQ25895M_REG0D = 0x0D;
+const BQ25895M_REG0E = 0x0E;
+const BQ25895M_REG0F = 0x0F;
+const BQ25895M_REG10 = 0x10;
+const BQ25895M_REG11 = 0x11;
+const BQ25895M_REG12 = 0x12;
+const BQ25895M_REG13 = 0x13;
+const BQ25895M_REG14 = 0x14;
+
+// For getChargeStatus() output
+enum chargingStatus{
+    not_charging, // 0
+    pre_charge, // 1
+    fast_charging, // 2
+    charge_termination_done // 3
+}
+
+// For CHGR_FAULT in getChargingFaults output
+enum chargingFault{
+    normal, // 0
+    input_fault, // 1
+    thermal_shutdown, // 2
+    charge_safety_timer_expiration // 3
+}
+
+// For NTC_FAULT in getChargingFaults output
+enum ntcFault{
+   normal, // 0
+   ts_cold, // 1
+   ts_hot // 2
+}
+
+class BQ25895M {
+    
+    static VERSION = "1.0.0";
+    
+    // I2C information
+    _i2c = null;
+    _addr = null;
+
+    constructor(i2c, addr=0x6a<<1){
+        
+        _i2c = i2c;
+        _addr = addr;
+        
+    }
+    
+    //PUBLIC METHODS
+    
+    // Initialize battery charger configuration registers 
+    function initCharger(){
+        
+        _setReg(BQ25895M_REG02, 0xf3); // Enable ADC
+        _setReg(BQ25895M_REG03, 0x3a); // Enable charger and set defaults
+        _setReg(BQ25895M_REG07, 0x8d); // Set defaults
+
+    }
+    
+    // Set the enable charging bit, charging will happen automatically
+    function enableCharging(){
+        
+        local rd = _getReg(BQ25895M_REG03);
+        rd = rd | (1 << 4); // set CHG_CONFIG bit
+        
+        _setReg(BQ25895M_REG03, rd);
+        
+    } 
+     
+    // Clear the enable charging bit, device will not charge until enableCharging() is called again
+    function disableCharging(){
+        
+        local rd = _getReg(BQ25895M_REG03);
+        rd = rd & ~(1<<4); // clear CHG_CONFIG bits
+        
+        _setReg(BQ25895M_REG03, rd);
+        
+    } 
+    
+    // Set target battery voltage
+    function setChargeVoltage(VREG){ 
+        
+        // Check that input is within accepted range
+        if (VREG < 3.840) VREG = 3.840;
```

Comment:
style note: unless it is a single line if statement curly braces should be used.
if (VREG < 3.840) {
     VREG = 3.840;
} else if (VREG < 4.608) {
    VREG = 4.608
}